jlab-sensing/soil-power-sensor-firmware
# Add support for CI linting, tests and remote development

**Name/Affiliation/Title**
John, UCSC, Maintainer

**Purpose of the PR**
This PR adds linting and static code analysis across project (stm32, esp32, and python) and integrates with the CI. Tests for python and stm32 are also integrated. This also enables remote development to be performed on the stm32 project.

**Development Environment**
`Linux spruce 6.8.2-arch2-1 #1 SMP PREEMPT_DYNAMIC Thu, 28 Mar 2024 17:06:35 +0000 x86_64 GNU/Linux`
PlatformIO Core, version 6.1.7
Used ST-Link V3 MINIE

**Test Procedure**
- Check CI actions to view the lint/static/test outputs
- Run through examples described in `docs/remove-dev.md`

**Additional Context**
There are some changes to `platformio.ini` where the custom dependency for `openocd` has been removed allowing people to use platformio binary versions.

Remote tests for the esp32 are not setup. See #109 for progress on the issues.

**Task List**


**Relevant Issues**
Closes #59
